### DISCUSSION TITLE
Set tsserver/node_modules location

### DISCUSSION THREAD

**Discussion by kenmc411** [mickaelistria/eclipse-bluesky#1489] [Q&A]
Hello, I have a hybrid Angular/Java project that contains a node_modules folder which uses typescript version 5.1.6.  However, the language server is pointing to the wild web developer default node_modules location per the Error log info line: 

![image](https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/assets/162622626/ec9862ba-53ed-4b49-9ed7-4296f426e57a)

My related Eclipse preference is set to use the typescript project version:

![image](https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/assets/162622626/41629fe9-68c2-4816-844d-e270e7f6c687)

Occasionally, if I kill the node processes for the language server, the info line in the error log will point to my project's node_modules directory.  I'm not sure why, but it does happen on rare occasion.  

![image](https://github.com/eclipse-wildwebdeveloper/wildwebdeveloper/assets/162622626/5ced9519-2ad6-4787-9c38-04e5cd5e9c54)

**I'd like to try pointing the typescript language server to always point to my project's node_modules, hopefully alleviating some problems I'm seeing using the 5.2.2 version ( I'm getting almost constant exceptions and all related features stop working). **

Installation Related Details
Windows 11

Eclipse 2023-12 -   Eclipse Platform	4.30.0.v20231201-0512	org.eclipse.platform.feature.group	Eclipse.org

Plugin Details
  Wild Web Developer HTML, CSS, JSON, Yaml, JavaScript, TypeScript, Node tools	1.3.3.202311211923	org.eclipse.wildwebdeveloper.feature.feature.group	Eclipse Wild Web Developer project
    Node.js embedder from Wild Web Developer	1.0.7.202310251358	org.eclipse.wildwebdeveloper.embedder.node.feature.feature.group	Eclipse Wild Web Developer project
    Wild Web Developer XML tools	1.3.1.202310310352	org.eclipse.wildwebdeveloper.xml.feature.feature.group	Eclipse Wild Web Developer project

**Comment by mickaelistria** [comment]
> **I'd like to try pointing the typescript language server to always point to my project's node_modules, hopefully alleviating some problems I'm seeing using the 5.2.2 version ( I'm getting almost constant exceptions and all related features stop working). **

This is unfortunately not possible yet. And it would be hard to take a reliable decision of what's best between Wild Web Developer default (which is known to be compatible) and the project module (which might be better). Sticking to default seems more reliable, so one would need to add to Wild Web Developer an option to load the module from a specific project if it's more recent and also compatible.
A contribution to add that would be very welcome.

**Reply by kenmc411** [reply]
I may be conflating a couple of items, but what does setting the WWD's preference for typescript to "project version" do? Or, what is it expected to do?  My project is 5.1.6, but the language server usually shows 5.2.2.  Looking through the WWD source code, I don't see where the preference is actually set to "Project version." I only see it set to "Eclipse Version" in the JSTSPreferenceServerConstants class.
